### ISSUE TITLE
Equality for unmanaged objects

### ISSUE THREAD

**Issue by roman-qp** [realm/realm-swift#7950]
### Problem

At the moment, the `RLMObjectBaseAreEqual` function always assumes that `RLMObjectBase` aren't equal if **one** of the objects has no realm associated. This may seem a bit confusing, because two unmanaged realm entities having the same properties, by logic, should be evaluated as equal.

![CleanShot 2022-09-08 at 13 23 37](https://user-images.githubusercontent.com/106338905/189218737-84418c51-458a-45be-967d-2556f9779de9.png)

### Solution

I think one of the solutions might be defaulting to the `NSObject.isEqual:(id)` implementation if both objects are unmanaged.

### How important is this improvement for you?

Would be a major improvement

**Comment by tgoyne** [comment]
We already match what `-[NSObject isEqual:]` does for unmanaged objects. We (and NSObject) use reference equality, and two objects which happen to have the same value are not equal because they are different objects.

**Comment by roman-qp** [comment]
> We already match what `-[NSObject isEqual:]` does for unmanaged objects. We (and NSObject) use reference equality, and two objects which happen to have the same value are not equal because they are different objects.

Good to know. I guess the only way to compare ununmanaged objects is to write a custom equalizer that will compare properties?

**Comment by tgoyne** [comment]
Yes, if you want memberwise equality checking you have to implement that manually. Note that you'll also need to override `hash` if you override `isEqual:`.
